MetaMask/metamask-extension
# dev(chore): log test names and suites

## **Description**
Before #21441 the test reporter would output the test path it was executing which made figuring out where tests were failing at fairly easy. The updates in #21441 make tests run faster but it doesn't help when trying to ascertain where failures occur. To make it slightly easier this PR adds some reporting to the run-all script and the withFixtures and verboseReportOnFailure functions to explicitly call out where breakdowns happen.

This PR also changes from using `this.test.title` to `this.test.fullTitle()` which includes the name spacing of Describe Blocks for easier access. 

## **Related issues**


## **Manual testing steps**

1. View CircleCI output on develop for any of the E2E tests.
2. Observe that the test steps are written out verbosely, but no information about what file or test is being ran.
3. Do the same on this branch's CircleCI output and see the the test files and names are logged.
4. Repeat steps locally on develop and this branch to see local terminal console outputs file paths and test names. 

## **Screenshots/Recordings**



### **Before**
<img width="1271" alt="Screenshot 2023-10-26 at 1 48 27 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/9dae6c6c-9c6b-4fc6-8852-c02ae7662c65">

### **After**
<img width="1249" alt="Screenshot 2023-10-26 at 2 15 21 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/fa6058f9-31b5-4dc7-b0fc-d9a76ca92cb8">
<img width="1197" alt="Screenshot 2023-10-26 at 2 15 41 PM" src="https://github.com/MetaMask/metamask-extension/assets/4448075/c5be627a-34fe-4e70-af1a-b478b0d06b4a">



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
